src/python/pants/backend/python/typecheck/mypy/subsystem.py

```diff
@@ -165,6 +165,12 @@ class MyPy(PythonToolBase):
             Alternatively, you can use this option so that the dependencies are solely
             used when running MyPy and are not runtime dependencies.
 
+            NOTE: Dependencies specified in this way are not visible to dependency inference,
+            and cannot be referenced as explicit dependencies. If you `import` from a stubs
+            module specified here, you may see warnings/errors from Pants depending on inference
+            settings. Specifying the dependencies as "typical" requirements via `requirements.txt`
+            or `python_requirement` targets avoids this issue.
```

Comment:
Alternatively, you can set `# pants: no-infer-dep` on the import line.

Reply:
I worry about recommending ^^^ without also saying it's only "safe" to do so within an `if typing.TYPE_CHECKING` guard, which is awkward to say without showing examples code... and I'm not sure how wordy we want help text to be. What do you think?

Reply:
Maybe actually this speaks to an improvement in the NOTE:

> If you `import` from a stubs module specified here—and it does not have a corresponding implementation `python_requirement` target that provides the imported module—then you may see...

Usually this isn't a problem! Like `requests-types` and `requests` works fine. Sounds like it's boto3 that is an issue

Reply:
Yeah boto3's stubs are particularly bad (the google-cloud type stubs are also weirdly named / don't match stub conventions, funnily enough). I can do some more word-smithing 🤔

Reply:
Re: having too much text overall, I had thought that maybe the help text here should link to the other docs so everything stays in one place. Is that a thing we do? Or best to keep the help text self-contained?

Reply:
It is a thing we do via the helper function `doc_url()` - grep for some examples